机器学习算法要求数据具有某种格式，然后才能对数据进行算法训练。 在本节中，我们对将要用于Sklearn库训练机器学习算法的数据，需要进行一些预处理的步骤。如把数据分为训练数据和测试数据，转化文本格式的数据为数值格式等。从上一节，我们可以使用pandas库从 CSV 文件中读取数据。 本节我们将采取另外一种方式，在Seaborn库中直接加载默认的数据集（本节内容要先安装seaborn库）。Seaborn库自带了一些数据集，要查看所有数据集目录，可以使用 get_dataset_names() 函数：

In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns

sns.get_dataset_names()

使用load_dataset()函数加载指定的数据集，下面的代码加载了tips数据集并显示前5行：

In [4]:
tips_data = sns.load_dataset("tips")
print(tips_data.head())
print(tips_data.shape)

   total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4
(244, 7)


该数据集一共有244条记录，每条记录共7项，记录了吃完饭后给小费的情况。7个数据项的含义依次是：账单总额、顾客给的小费额、性别、在吃饭时是否抽烟、吃饭的星期、时间（晚餐或日间餐）、聚餐人数。

接下来，我们需要把数据集分成“特征”和“标签”2部分。由于我们将根据数据建立模型，并且最终的目的是预测顾客吃完饭后会给多少钱的小费。因此，我们将小费这一列作为“标签”，而其他剩余的列则作为“特征”。下来将数据集分为2个：

In [7]:
X = tips_data.drop(['tip'], axis=1)
Y = tips_data['tip']

0    1.01
1    1.66
2    3.50
3    3.31
4    3.61
Name: tip, dtype: float64

在大多数情况下，机器学习算法只能处理数字。 因此，将分类数据转换为数字格式很重要。

在这方面，第一步是创建所有数值的数据集。 为此，请从数据集中删除分类列，如下所示。